website/docs/cloud-docs/workspaces/settings/vcs.mdx

```diff
@@ -4,82 +4,63 @@ page_title: VCS Connections - Workspaces - Terraform Cloud and Terraform Enterpr
 
-  For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repo. You can add `modules` as a trigger directory for each workspace to make sure they notice any changes to shared code.
-- **Mark the entire repository as relevant.** If you set the "Automatic Run Triggering" setting to "Always Trigger Runs," Terraform Cloud will assume that anything in the repository might affect the workspace's configuration, and will queue runs for any change.
+#### Only trigger runs when files in specified paths change
 
-  This can be useful for repos that don't have multiple configurations but require a working directory for some other reason. It's usually not what you want for true monorepos, since it queues unnecessary runs and slows down your ability to provision infrastructure.
+This option instructs Terraform to begin new runs only for changes that affected specified files and directories. This behavior also applies to [speculative plans](/cloud-docs/run/#speculative-plans) on pull requests.
 
-### Error Handling for Run Triggers
+You can use trigger patterns and trigger prefixes in the **Add path** field to specify groups of files and directories.
+- **Trigger Patterns:** (Recommended) Use glob patterns to specify the files that should trigger a new run. For example, `/submodule/**/*.tf`, specifies all files with the `.tf` extension that are nested below the `submodule` directory. You can also use more complex patterns like `/**/networking/**/*`, which specifies all files in the `networking` folder at this level of the file path (e.g., `/submodule/service-1/networking/private/main.tf`). Refer to [TODO: create glob patterns page](link) for details.
+- ** Trigger Prefixes:** Terraform Cloud will queue runs for changes in any of the specified trigger directories matching the provided prefixes (including the working directory). For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repo. You can add `modules` as a trigger directory for each workspace to make sure they notice any changes to shared code.
 
-Terraform Cloud retrieves the changed files for each push or pull request using your VCS provider's API. If for some reason the list of changed files cannot be retrieved, or if it is too large to process, the default behavior is to trigger runs on all attached workspaces. Should this happen, you may see several runs with state "Planned", due to the push resulting in no changes to infrastructure.
+-> **Note:** Terraform triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.
```

Comment:
@matejrisek I don't know that we needed an entire subsection for the information about what happens if terraform can't retrieve the list of changes. So I tried putting it in a note. Thoughts?   

Also, I don't really understand what the current content means. To "All attached workspaces" --> does that mean that if TFC can't get information about a repository, it automatically just triggers runs in any/all terraform cloud workspaces that are attached to that repository? How often does this happen.

Reply:
We should start with `Terraform Cloud` instead of `Terraform`, since it's TFC that triggers runs :)

Aside from that it looks good to me even though I'm not completely familiar with the behavior of TFC in this case. But I'd say the semantics would have stayed the same after this change - so it's good to go.